bdns/servers.go

```diff
@@ -113,23 +119,77 @@ type dynamicProvider struct {
 	updateCounter *prometheus.CounterVec
 }
 
+// resolveDNSAuthority resolves the DNS authority to use for resolution of other
+// DNS backends. The DNS authority can be specified as a hostname or IP address,
+// with or without a port. If the authority is specified as a hostname it will
+// be resolved via the system DNS. If the authority is specified as an IP
+// address it will be used directly, defaulting to port 53 if no port is
+// specified.
+func resolveDNSAuthority(d string) (string, error) {
+	host, port, err := net.SplitHostPort(d)
+	if err != nil {
+		// Assume host with no port specified, default port to 53.
+		host = d
+		port = "53"
+	} else {
+		// Ensure the `port` portion of `address` is a valid port.
+		portNum, err := strconv.Atoi(port)
+		if err != nil {
+			return "", errors.New("port must be an integer: %s")
```

Comment:
```suggestion
			return "", fmt.Errorf("parsing port number: %s", err)
```

(I wonder if there's a lint available for "formatting directive in errors.New")

Reply:
Fixed here and where I copied it from, above.